python/analyze.py

```diff
@@ -5,46 +5,121 @@
 import librosa.display
 import pandas as pd
 from tqdm import tqdm
+import matplotlib.pyplot as plt
+import sklearn
 
 
-def get_mfcc(dir):
+
+def get_features(dir):
+    sr = 22050
+    duration = 5
     try:
-        audio, srate = librosa.load(dir, res_type='kaiser_fast')
-        nffts = [2048, 1024, 512, 256, 128]
-
-        # Ensures that the fft will have adequate frame size
-        for i in nffts:
-            if i <= audio.size:
-                nfft = i
-        nffts = 256
+        audio, srate = librosa.load(dir, sr = sr, mono = True, duration = duration, res_type='kaiser_fast')
+        audio, index = librosa.effects.trim(y = audio, top_db=60, frame_length=2048, hop_length=512)
+        nfft = 2048
+        hop = 1024
+
+        # 0 pad to 2048 minimum size
+        if audio.size < 2048:
+            audio = np.pad(audio, (0, 2048 - audio.size), constant_values=(0, 0))
```

Comment:
I think use 1024 for hop length for the trim hop length too. I was initially thinking a shorter hop length would let us trim more but we would have to reduce the frame size to get that effect. For the sake of speed lets use 1024. 

Also, all these instances of 2048 should probably be referencing `nfft` and `hop`. I would move those above the call to trim to avoid repeating number literals in the code.